co-log/example/Main.hs

```diff
@@ -1,26 +1,30 @@
-{-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications  #-}
 
 module Main where
 
-import Colog (WithLog, cmap, logMsg, logStringStderr, logStringStdout, usingLoggerT, withLog,
-              withLogStringFile)
+import Colog (LogMessage (..), WithLog, cmap, fmtLogMessage, logDebug, logInfo, logMsg,
+              logTextStderr, logTextStdout, logWarning, usingLoggerT, withLog, withLogTextFile)
 
-example :: WithLog env String m => m ()
+example :: WithLog env LogMessage m => m ()
 example = do
-    logMsg "First message..."
-    logMsg "Second message..."
+    logDebug "First message..."
+    logInfo "Second message..."
 
-app :: WithLog env String m => m ()
+app :: WithLog env LogMessage m => m ()
 app = do
-    logMsg "Starting application..."
-    withLog (cmap ("app:" ++)) example
+    logWarning "Starting application..."
+    withLog (cmap addApp) example
+  where
+    addApp :: LogMessage -> LogMessage
+    addApp (LogMessage msg sev) = LogMessage ("app: " <> msg) sev
```

Comment:
That's the correct way to handle `cmap` for data types. But with lens this can be written shorter without need to introduce new function each time, like:

```haskell
withLog (cmap $ msg %~ ("app:" <>)) example
```

(though, some people might say that the above code is not really readable)

I think that we will add some lens library to `colog` at some point in future. I haven't decided yet, which one: `lens` or `microlens`. But we will figure out it later.